service/raft/raft_group0.cc

```diff
@@ -366,6 +366,8 @@ future<group0_info> persistent_discovery::run(
 }
 
 future<> raft_group0::abort() {
+    co_await stop_group0();
+
     co_await smp::invoke_on_all([this]() {
```

Comment:
RPC verbs are initialized first, *then* group 0 server is started.

The shutdown order should be the reverse of the init order.

In this case, imagine one of the RPC verbs accesses the group 0 server. If we stop and remove the server first, a verb handler may fire which depends on the server, trying to access a removed server.

In fact `stop_group0` should be the last step of `abort` -- after `_shutdown_gate.close()` -- the gate is held by tasks that are accessing the server too. So we should first close the gate, ensuring that none of these tasks are running, and only then stop the server.